server/admin_management/api/app/models/model.py

```diff
@@ -166,6 +168,75 @@ class FamApplicationAdmin(Base):
     )
 
 
+class FamAccessControlPrivilege(Base):
+    __tablename__ = "fam_access_control_privilege"
+    __table_args__ = (
+        ForeignKeyConstraint(
+            ["role_id"],
+            ["app_fam.fam_role.role_id"],
+            name="reffam_access_control_privilege_role",
+        ),
+        ForeignKeyConstraint(
+            ["user_id"],
+            ["app_fam.fam_user.user_id"],
+            name="reffam_access_control_privilege_user",
+        ),
+        PrimaryKeyConstraint(
+            "access_control_privilege_id", name="fam_access_control_privilege_pk"
+        ),
+        UniqueConstraint("user_id", "role_id", name="fam_access_control_usr_rle_uk"),
+        Index("ix_app_fam_fam_access_control_privilege_role_id", "role_id"),
+        Index("ix_app_fam_fam_access_control_privilege_user_id", "user_id"),
+        {
+            "comment": "Access Control Privilege is a cross-reference object that allows "
+            "for the identification of who are the delegated "
+            "administrators(User) for an Application for a particular role.",
+            "schema": "app_fam",
+        },
+    )
+    access_control_privilege_id = Column(
+        BigInteger,
+        Identity(
+            start=1,
+            increment=1,
+            minvalue=1,
+            maxvalue=9223372036854775807,
+            cycle=False,
+            cache=1,
+        ),
+        primary_key=True,
+        comment="Automatically generated key used to identify the uniqueness of a User administers the Application role.",
+    )
+    user_id = Column(
+        BigInteger,
+        comment="Unique ID to reference and identify the user within FAM system.",
+    )
+    role_id = Column(
+        BigInteger,
+        comment="Unique ID to reference and identify the application role within FAM system.",
+    )
+    create_user = Column(
+        String(60), comment="The user or proxy account that created the record."
+    )
+    create_date = Column(
+        TIMESTAMP(timezone=True, precision=6),
+        nullable=False,
+        default=datetime.datetime.utcnow,
+        comment="The date and time the record was created.",
+    )
+    update_user = Column(
+        String(60),
+        comment="The user or proxy account that created or last updated the record.",
+    )
+    update_date = Column(
+        TIMESTAMP(timezone=True, precision=6),
+        onupdate=datetime.datetime.utcnow,
+        comment="The date and time the record was created or last updated.",
+    )
+    role = relationship("FamRole", back_populates="fam_access_control_privilege")
+    user = relationship("FamUser", back_populates="fam_access_control_privilege")
```

Comment:
Hi Ian, it's just my for my learning, I wonder when we need lazy="join"?

Reply:
Hi Catherine, that's good question though!
On high level, you might already know, it means "lazy loading" from ORM objects.
When there is a relationship (parent/child) between two ORM model classes, this "lazy" gives choices for when to load the relationship (child) object(s) when you have a select statement. And for loading relationship, there are 3 types: Eager, Lazy, or None (not loading at all). "Lazy" mode of loading means the ORM engine will only load the relationship when you actually access the child objects. 

However, in the beginning when learning to code in Python with SQLAlchemy, honestly I was aware but didn't pay much attention to it because SQLAlchemy made it quite confusing to me for some reason (than Java JPA). But since you asked, I looked again to its docs, still confusing because SQLAlchemy makes this "lazy" on many types. 
I think, your question on lazy="join" to me is essentially an "Eager" as mentioned one of 
the 3 types above (although it has lazy/join wording, to me it means lazy="false").

In this case on your commented area, the relationships to "user" and "role" have no "lazy" in the mapping, so I believe it is by default Lazy="true". In the past, very often this (lazy="true") is good for performance and probably what we want; Eager loading will be a little slower. However, I am not really sure in Python, where the ORM session boundary is...so that it can still manage the loading strategy...
I would say, leave it as default, then when we have CRUD code, pay attention to the query. If it does not return the child relationship we want, then probably we need to turn it to Eager mode (easiest way) or know where the lazy loading boundary is (normally it ends within the function being called, but I am not sure at Python, maybe more to learn in SQLAlchemy in detail).

Reply:
Thanks Ian!! yes I was checking the documentation, and the lazy has a few options, but the explanation is not very clear. I like the idea to check if when we use it. Cause I saw the application admin model uses the lazy joint. thanks so much!!

Reply:
Thanks Catherine, I don't remember in application admin model why it has that... maybe was inherited from other model, we can tweak when we know more later.